service_storagefacility/app/services/StorageNodeService.scala

```diff
@@ -515,45 +514,45 @@ class StorageNodeService @Inject() (
   ): Future[MusitResult[Seq[StorageNodeDatabaseId]]] = {
     // Calling get on affectedThing, after filtering out nonEmpty ones, is safe.
     val nodeIds = moveEvents.filter(_.affectedThing.nonEmpty).map(_.affectedThing.get) // scalastyle:ignore
-    val affectedNodes = unitDao.getNodesByIds(mid, nodeIds)
-    val currLoc = affectedNodes.map(_.map(n => (n.id.get, n.isPartOf)).toMap)
-
-    logger.debug(s"Preparing to move nodes to $destination")
-
-    moveBatch(mid, destination, nodeIds, currLoc, moveEvents) {
-      case (to, curr, events) =>
-        logger.debug(s"Destination node is ${to.id} with path ${to.path}")
-        logger.debug(s"Filtering away invalid placement of nodes in $destination.")
-        // Filter away nodes that didn't pass first round of validation
-        val nodesToMove = affectedNodes.map { nodes =>
-          nodes.filter(n => events.exists(_.affectedThing.contains(n.id.get)))
-        }
 
-        // Filter away nodes with invalid positions and process the ones remaining
-        filterInvalidPosition(mid, to.path, nodesToMove).flatMap { validNodes =>
-          if (validNodes.nonEmpty) {
-            logger.debug(s"Will move ${validNodes.size} to $destination.")
-            // Remove events for which moving the node was identified as invalid.
-            val validEvents = events.filter(e => validNodes.exists(_.id == e.affectedThing))
-            val moveIds = validNodes.flatMap(_.id)
-
-            for {
-              // Update the NodePath and partOf for all nodes to be moved.
-              resLocUpd <- unitDao.batchUpdateLocation(validNodes, to)
-              if resLocUpd.isSuccess
-              // If the above update succeeded, we store the move events.
-              mvRes <- persistMoveEvents(mid, validEvents)(_ => MusitSuccess(moveIds))
-            } yield {
-              logger.debug(s"Successfully moved ${validNodes.size} nodes to $destination")
-              mvRes
-            }
-          } else {
-            Future.successful {
-              MusitValidationError("No valid commands were found. No nodes were moved.")
+    val res = for {
+      affectedNodes <- MusitResultT(unitDao.getNodesByIds(mid, nodeIds))
+      currLoc = affectedNodes.map(n => (n.id.get, n.isPartOf)).toMap
+      movedObjects <- MusitResultT(moveBatch(mid, destination, nodeIds, currLoc, moveEvents) { // scalastyle:ignore
```

Comment:
Right...this is quite big...I've made a change here where I extract the stuff inside the case(to, cure, events) bit into a private function.